docs/api_specs/includes/watcher_api_specs.md

```diff
@@ -0,0 +1,691 @@
+> A piggyback_available event
+
+```json
+{
+    "event": "piggyback_available",
+    "details": {
+        "inflight_blocknum"  : 10000,
+        "inflight_tx_index"  : 12,
+        "inflight_io_index"  : 1
+    }
+}
+```
+
+An in-flight exit has been started and can be piggybacked 
+<aside class="warning"> Not Implemented Yet.</aside> 
+
+Event details:
+Attribute | Type | Description
+--------- | ------- | -----------
+inflight_blocknum | integer | Block number of the in-flight transaction
+inflight_tx_index | integer | Index of the in-flight transaction
+inflight_io_index | integer | Index of the input or output that can be exitted.
+
+#### `invalid_piggyback`
+> A invalid_piggyback event
+
+```json
+{
+    "event": "invalid_piggyback",
+    "details": {
+        "type": "output",
+        "inflight_blocknum"  : 10000,
+        "inflight_tx_index"  : 12,
+        "inflight_io_index"  : 1,
+        "spending_blocknum"  : 10000,
+        "spending_tx_index"  : 10,
+        "spending_input_index"  : 0,
+    }
+}
+```
+
+An invalid piggyback is in process. Should be challenged.
+<aside class="warning"> Not Implemented Yet.</aside>
+
+Event details:
+Attribute | Type | Description
+--------- | ------- | -----------
+type | string | Indicates whether the invalid piggyback is on an "input" or "output"
+inflight_blocknum | integer | Block number of the in-flight transaction
+inflight_tx_index | integer | Index of the in-flight transaction
+inflight_io_index | integer | Index of the input or output of the in-flight transaction
+spending_blocknum | integer | Block number of the spending transaction
+spending_tx_index | integer | Index of the spending transaction
+spending_input_index | integer | Index of the spent input iofn the spending transaction
+
+
+## Inflight Exit - Get Exit Data
+
+```shell
+curl http://localhost:4000/inflight_exit.get_data -d '{"txhash": "bdf562c24ace032176e27621073df58ce1c6f65de3b5932343b70ba03c72132d"}'
+```
+
+```elixir
+// TODO
+```
+
+```javascript
+// TODO
+```
+
+> The above command returns JSON document:
+
+```json
+{
+    "version": "1",
+    "success": true,
+    "data": {
+        "txbytes": "F847010180808080940000...",
+        "sigs": "7C29FB8327F60BBFC62...",
+        "input_txs" : [
+            {
+                "txbytes": "F81891018080808...",
+                "proof": "CEDB8B31D1E4C..."
+            },
+            {
+                "txbytes": "2A03418086001...",
+                "proof": "A67131D1E4C..."
+            }
+        ]
+    }
+}
+```
+
+Gets exit data for an in-flight exit
+
+### HTTP Request
+
+`POST /inflight_exit.get_data`
+
+### Request Body
+
+Attribute | Type | Description
+--------- | ------- | -----------
+txhash | Hex encoded string | The hash of the in-flight transaction
+
+
+
+## Inflight Exit - Get Challenge Data
+
+```shell
+curl http://localhost:4000/inflight_exit.get_challenge_data -d '{"txhash": "bdf562c24ace032176e27621073df58ce1c6f65de3b5932343b70ba03c72132d"}'
```

Comment:
similar here - how does this correlate with what we're getting from `status`? how would the user's flow be satisfied?

Hm.. in general I think it could be worthwhile to write out all of the `ACTION:...` entries from the MoreVP handling flows, in detail. E.g.:

1/ user pings `status/`
2/ user takes some data out of the response
3/ optionally user makes some decision based on that data
4/ user calls some endpoint using some piece of that data
5/ user turns the data from that response into an ethereum tx and calls the contract

That would allow us to double check whether all the actions are possible

Reply:
I've added this in inflight_exit_scenarios.md, which has led me to a few changes in the approach

1. /status returns a list of `byzantine_events` and also a list of `inflight_txs` and `inflight_exits`. I think separating them is cleaner and it avoids duplicating information in each event.
2. Events are related to the transactions by `txbytes`.  This is maybe a too heavy and could be done on `txhash` instead. But `txbytes` is used almost everywhere.
3. I'm assuming the Watcher can track piggybacks. For example it will report if an output is available to be piggybacked. If that output does get piggybacked then it won't report it as available anymore. Not sure if this is actually feasible?

Reply:
@pik694 @purbanow @paulperegud please, take a look at the bit `txbytes` vs `txhash` in Kevin's docs vs how you're solving on your side